src/transport_skarab.py

```diff
@@ -1284,109 +1284,125 @@ def unpack_payload(response_payload, response_type, number_of_words,
         # return response from skarab
         return SkarabTransport.sd_dict[response_type](*unpacked_data)
 
-    @property
-    def seq_num(self):
-        return self._seq_num
-
-    @seq_num.setter
-    def seq_num(self, value):
+    def update_seq_num(self):
         self._seq_num = 0 if self._seq_num >= 0xffff else self._seq_num + 1
 
-    def send_packet(self, payload, response_type,
-                    expect_response, command_id, number_of_words,
+    def send_packet(self, request_object, response_type,
+                    expect_response, number_of_words,
                     pad_words,
                     timeout=sd.CONTROL_RESPONSE_TIMEOUT,
-                    retries=3,
+                    retries=5,
                     skarab_socket=None, port=None):
         """
         Send payload via UDP packet to SKARAB
         Sends request packets then waits for response packet if expected
-        Retransmits request packet (up to 3 times) if response not received
+        Retransmits request packet if response not received
 
         :param skarab_socket: socket object to be used
-        :param port:
-        :param payload: the data to send to SKARAB
+        :param port: port to connect to
+        :param request_object: object containing the data to send to SKARAB
         :param response_type: type of response expected
         :param expect_response: is a response expected?
-        :param command_id: command_id of the request packet
         :param number_of_words: total number of 16-bit words expected in response
         :param pad_words: number of padding words (16-bit) expected in response
-        :param timeout
-        :param retries
+        :param timeout: how long to wait for a response before bailing
+        :param retries: how many times to retransmit a request
         :return: response expected: returns response object or 'None' if no
-        response received. else returns 'ok'
+        response received.
         """
         # TODO - refactor the requests/responses into one
-        # TODO - if the packet payloads are being formed here, we don't need to pass the sequence number to every Command. Rather the command that makes the payload should take the sequence number.
-        # TODO - Retries should use a different sequence number. Can't do it here, since sequence number is added when payload is generated. Needs refactoring.
 
         # default to the control socket and port
         skarab_socket = skarab_socket or self.skarab_ctrl_sock
         port = port or self.skarab_eth_ctrl_port
+
         retransmit_count = 0
         while retransmit_count < retries:
             LOGGER.debug('Retransmit attempts: {}'.format(retransmit_count))
             try:
+
+                if retransmit_count != 0:
```

Comment:
For me, this would make more sense at the end of the loop. So if you're going to retry, update the sequence number and try again. So the normal execution is always the same and you don't need to do this check.